### DISCUSSION TITLE
PasswordBox - Bindable?

### DISCUSSION THREAD

**Discussion by msavazzi** [MahApps/MahApps.Metro#4043] [Q&A] [answered]
Team,
does Mahapps provide a method to bind the Passoword property?
I've not seen it in the example.

Here the project I've tried but it ends always with an empty password:
[https://github.com/msavazzi/Mahapps.PasswordBind](https://github.com/msavazzi/Mahapps.PasswordBind)

**Comment by makingbloke** [comment, marked as answer]
Yes it does have. Here's an example from a project of mine:

`        <PasswordBox mahControls:PasswordBoxHelper.CapsLockWarningToolTip="Caps Lock is on" mahControls:TextBoxHelper.Watermark="Password" mahShared:PasswordBoxBindingBehavior.Password="{Binding Password, UpdateSourceTrigger=PropertyChanged, ValidatesOnDataErrors=True, Mode=TwoWay}" />
`
Notice the MahApps PasswordBoxBindingBehaviour is used to bind the password to a Password string property in my ViewModel.

Just FYI the XAML namespaces are:

`    xmlns:mahControls="http://metro.mahapps.com/winfx/xaml/controls"
    xmlns:mahShared="http://metro.mahapps.com/winfx/xaml/shared"
`

**Reply by msavazzi** [reply]
@makingbloke I have a major issue. When I leave the settings page and go back to the first page the password got cancelled.
I've added some debug strings and here the outcome:

I enter the page:
> BackupPassword >test<

I press 2 and add it
> BackupPassword >test2<

When I press on the main page
> BackupPassword ><
> navigated from

So the password is always empty.
If I change the passwordBox into a textbox that does not happen

**Reply by timunie** [reply]
@msavazzi your demo is missing very important parts. Please double-check it.

**Comment by makingbloke** [comment]
Sorry, I don’t know, sounds more like something in your project. I’m using
MVVM and the password control is bound to a property in the View Model. The
VM is a singleton that gets injected into the logon form so I can get at
the password by looking at this.

**Reply by msavazzi** [reply]
@makingbloke I've created a sample project here: https://github.com/msavazzi/Mahapps.PasswordBind with nothing in it apart your suggestion or a potential other solution. Both of them does not work. 
can you be so kind to check what I'm doing wrong?

**Reply by timunie** [reply]
> @msavazzi your demo is missing very important parts. Please double-check it.

**Comment by timunie** [comment]
Hi @msavazzi 

I provided an working solution to your sample (different ViewModel as the one you provided does not run). The issue must be something else, but we are not here to solve all your issues. See: https://github.com/msavazzi/Mahapps.PasswordBind/pull/2 

/cc @makingbloke fyi 

Happy coding
Tim
